src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -42,39 +42,52 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
                     throw ExceptionUtil.ParameterInvalid(nameof(exePath));
                 }
-
-                applicationFilename = ApplicationUri;
             }
             else
             {
                 // Exe path wasn't specified, get it from the entry assembly
                 exeAssembly = Assembly.GetEntryAssembly();
 
-                if (exeAssembly == null)
-                    throw new PlatformNotSupportedException();
-
-                HasEntryAssembly = true;
+                if (exeAssembly != null)
+                {
+                    HasEntryAssembly = true;
 
-                // The original .NET Framework code tried to get the local path without using Uri.
-                // If we ever find a need to do this again be careful with the logic. "file:///" is
-                // used for local paths and "file://" for UNCs. Simply removing the prefix will make
-                // local paths relative on Unix (e.g. "file:///home" will become "home" instead of
-                // "/home").
-                string configBasePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, exeAssembly.ManifestModule.Name);
-                Uri uri = new Uri(configBasePath);
+                    // The original .NET Framework code tried to get the local path without using Uri.
+                    // If we ever find a need to do this again be careful with the logic. "file:///" is
+                    // used for local paths and "file://" for UNCs. Simply removing the prefix will make
+                    // local paths relative on Unix (e.g. "file:///home" will become "home" instead of
+                    // "/home").
+                    string configBasePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, exeAssembly.ManifestModule.Name);
+                    Uri uri = new Uri(configBasePath);
 
-                if (uri.IsFile)
-                {
-                    ApplicationUri = uri.LocalPath;
-                    applicationFilename = uri.LocalPath;
+                    if (uri.IsFile)
+                    {
+                        ApplicationUri = uri.LocalPath;
+                    }
+                    else
+                    {
+                        ApplicationUri = Uri.EscapeDataString(configBasePath);
+                    }
                 }
                 else
                 {
-                    ApplicationUri = Uri.EscapeDataString(configBasePath);
+                    // An EntryAssembly may not be found when running from a custom host.
+                    // Try to find the native entry point.
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    {
+                        string moduleFileName = Interop.Kernel32.GetModuleFileName(new HandleRef(null, IntPtr.Zero));
+                        if (!string.IsNullOrEmpty(moduleFileName))
+                        {
+                            ApplicationUri = Path.GetFullPath(moduleFileName);
```

Comment:
Is it really necessary to call `GetFullPath` on what `GetModuleFileName` returns?